pytensor/link/pytorch/dispatch/basic.py

```diff
@@ -13,7 +13,12 @@
 @singledispatch
 def pytorch_typify(data, dtype=None, **kwargs):
     r"""Convert instances of PyTensor `Type`\s to PyTorch types."""
-    return torch.as_tensor(data, dtype=dtype)
+    if isinstance(data, NoneType):
+        raise NotImplementedError(f"pytorch_typify is not implemented for {type(data)}")
+    if isinstance(data, slice):
+        raise NotImplementedError(f"pytorch_typify is not implemented for {type(data)}")
+    else:
+        return torch.as_tensor(data, dtype=dtype)
```

Comment:
You're missing the idea. By default nothing is implemented:

```python
@singledispatch
def pytorch_typify(data, **kwargs):
  raise NotImplementedError(f"pytorch_typify is not implemente for {type(data)}")
```

Then you can have a specific dispatch for the cases that were previously covered by the default case

```python
@pytorch_typify.register(np.ndarray)
@pytroch.typify.register(torch.Tensor)
def pytorch_tipyfy_tensor(data, dtype=None, **kwargs):
  return torch.as_tensor(data, dtype=dtype)
```

Does that make sense?